app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -206,7 +206,8 @@ def setup_agent(instance)
     end
 
     # run docker image
-    command_line = "sudo docker run -d --restart=always -v /dev:/host_dev -v #{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml --privileged #{docker_image}"
+    image = docker_login_required? ? "#{docker_registry}/#{docker_image}" : docker_image
```

Comment:
@hsong-rh Is `docker_login_required?` the correct test here? I would think we' want to join them whenever `docker_registry` is not empty.

Reply:
Correct, the proper test is when docker_registry is `.present?`

Reply:
docker_registry is only needed when docker_login_required? is true. I feel checking docker_login_required? is easier to understand.

Reply:
> docker_registry is only needed when docker_login_required? is true

That's not true, registries other than docker hub don't always require authentication.

Reply:
Changed to check docker_registry.present? for combination.